WordPress/gutenberg
# E2E Tests: Migrate Comments block test to Playwright

## What?
Migrate Comments block e2e tests to Playwright.

## Why?
Playwright infrastructure was recently added to Gutenberg (as [covered](https://make.wordpress.org/core/2022/03/23/migrating-wordpress-e2e-tests-to-playwright/) on make.wp.org/core); here's a [list](https://github.com/WordPress/gutenberg/issues/38851) of all e2e tests that need migrating.

The Comments block e2e test was only added recently (#39502), so our memory is still fresh, making it a good candidate for migration 🙂 

## How?
Following migration instructions found [here](https://github.com/WordPress/gutenberg/blob/3dddb153c955ecb69336298e3384195c1ad47cdd/test/e2e/MIGRATION.md).

Specifically, we're introducing new utilities to `deleteAllComments` (via REST API) and to `setOption` (via `wp-admin`).

## Testing Instructions
Verify that the new e2e test passes (check CI). To test locally: 

```
npm run test-e2e:playwright -- test/e2e/specs/editor/blocks/comments.spec.js
```

To run in interactive mode, append `--headed`.

## TODO

- [x] Add utility to create new comment ([via REST API](https://developer.wordpress.org/rest-api/reference/comments/#create-a-comment)).
- [x] Use more semantic selectors/locators and assertions, if possible.
- [x] ~Add Changelog entries to `packages/e2e-test-utils-playwright`.~ _(Still unreleased, so not required)_
